client/client.go

```diff
@@ -490,7 +490,7 @@ func (c *simpleHTTPClient) Do(ctx context.Context, act httpAction) (*http.Respon
 		case ctx.Err() != nil:
 			err = ctx.Err()
 		case hctx.Err() != nil:
-			err = fmt.Errorf("client: endpoint %s exceeded header timeout", c.endpoint.String())
+			err = hctx.Err()
```

Comment:
actually we should not return context error here. this error is not caused by the context cancellation or deadline exceeded by the context user passed in at line 441. This is caused by the header timeout user set at line 438. They are different thing. 

/cc @heyitsanthony